migration-engine/core/src/cli.rs

```diff
@@ -119,6 +113,37 @@ fn create_conn(
     }
 }
 
+/// Try to connect as an admin to a postgres database. We try to pick a default database from which
+/// we can create another database.
+fn create_postgres_admin_conn(mut url: Url) -> crate::Result<SqlMigrationConnector> {
+    let candidate_default_databases = &["postgres", "template1"];
+
+    let mut params: HashMap<String, String> = url.query_pairs().into_owned().collect();
+    params.remove("schema");
+    let params = params.into_iter().map(|(k, v)| format!("{}={}", k, v)).join("&");
+    url.set_query(Some(&params));
+
+    let inner = candidate_default_databases
+        .iter()
+        .filter_map(|database_name| {
+            url.set_path(database_name);
+            match SqlMigrationConnector::postgres(url.as_str(), false) {
+                // If the database does not exist, try the next one.
+                Err(SqlError::DatabaseDoesNotExist { .. }) => None,
+                // If the outcome is anything else, use this.
+                other_outcome => Some(other_outcome),
```

Comment:
Maybe we should include authorization errors (it still means we can't connect to it).

Reply:
We already check that with `can_connect`.